Source/WebCore/page/NavigationTransition.h

```diff
@@ -25,27 +25,29 @@
 
 #pragma once
 
-#include "DOMPromiseProxy.h"
 #include "NavigationHistoryEntry.h"
 #include "NavigationNavigationType.h"
 #include "ScriptWrappable.h"
 
 namespace WebCore {
 
+class DOMPromise;
+
 class NavigationTransition final : public RefCounted<NavigationTransition>, public ScriptWrappable {
     WTF_MAKE_ISO_ALLOCATED(NavigationTransition);
 public:
-
-    using FinishedPromise = DOMPromiseProxy<IDLUndefined>;
+    static Ref<NavigationTransition> create(NavigationNavigationType type, Ref<NavigationHistoryEntry> fromEntry) { return adoptRef(*new NavigationTransition(type, fromEntry)); };
 
     NavigationNavigationType navigationType() { return m_navigationType; };
     NavigationHistoryEntry& from() { return m_from; };
-    FinishedPromise& finished() { return *m_finished; };
+    RefPtr<DOMPromise> finished() { return m_finished; };
```

Comment:
`DOMPromise* finished() { return m_finished.get(); };`

Reply:
Thanks for the review. I'm curious what the reason here would be to avoid `RefPtr`?